.pre-commit-config.yaml

```diff
@@ -13,18 +13,19 @@ repos:
       - id: requirements-txt-fixer
         exclude: docs/.*
   - repo: https://github.com/timothycrosley/isort
-    rev: 5.10.1
+    rev: 5.11.4
     hooks:
       - id: isort
   - repo: https://github.com/psf/black
     rev: 22.3.0
     hooks:
       - id: black
-  - repo: https://github.com/PyCQA/flake8
-    rev: 5.0.4
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 'v0.0.229'
     hooks:
-      - id: flake8
-        files: ^(great_expectations/core)
+      - id: ruff
+        files: ^(great_expectations/core)  # TODO: expand to all of gx
+        args: ["--fix"]
```

Comment:
Ruff can actually fix a fair number of issues.
For example, if it finds an `unused-import`, it will remove it.

Reply:
do we always use all of our imports? If I do an `import sqlalchemy as sa`, am I guaranteed to use `sa`? I'm not so confident about this but you've done the research and I guess you're right!

Reply:
No there are cases where we just need to "re-export" or raise up an import like in `__init__.py`.
For that ignore these rules for all `__init__.py` just like we did before in the `flake8` `setup.cfg` file.
Also if the import is declared in `__all__` it will also be ignored.

But just as further reassurance, if I missed one of these cases it would also fail the CI lint step. We don't run `--fix` anywhere in the CI.
The only way it would happen is with an installed pre-commit hook or manually running `invoke lint --fix` or `ruff . --fix`